bigtable-hbase-dataflow/src/test/java/com/google/cloud/bigtable/dataflow/CloudBigtableConnectionPoolTest.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.dataflow;
+
+import static org.junit.Assert.assertNotSame;
+import static org.junit.Assert.assertSame;
+import static org.mockito.Mockito.mock;
+
+import java.io.IOException;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.client.Connection;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import com.google.cloud.bigtable.dataflow.CloudBigtableConnectionPool.PoolEntry;
+
+/**
+ * Tests for {@link CloudBigtableConnectionPool}
+ */
+@RunWith(JUnit4.class)
+public class CloudBigtableConnectionPoolTest {
+
+  private static class TestCloudbigtableConnectionPool extends CloudBigtableConnectionPool {
+    private long expireTime = System.currentTimeMillis() + 100000;
+    @Override
+    protected PoolEntry createConnection(Configuration config, String key) throws IOException {
+      return new PoolEntry(key, mock(Connection.class), expireTime);
+    }
+  }
+  
+  @Test
+  public void testConnectionIsPooled() throws IOException{
+    TestCloudbigtableConnectionPool pool = new TestCloudbigtableConnectionPool();
+    Configuration config = new Configuration();
+    PoolEntry entry = pool.getConnection(config, "key");
```

Comment:
It seems that we don't need line 50 and 51.
If we can verify the pool size, it be nice.